### PR TITLE
Vector search phase 1

### PR DESCRIPTION
**Pull request by gilkal**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- Add the tests for CBL vector-search 
- They are currently skipped, until all the test apps changes for all the platforms are merged.
- Tested on 3 platforms with a full API run.

### CODE REVIEW

File: testsuites/CBLTester/CBL_Functional_tests/APITests/test_vectorSearch.py [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+            scope: {
+                "collections": {
+                    def_col: {"sync": sync_function},
+                    st_col_name: {"sync": sync_function},
+                    dbv_col_name: {"sync": sync_function},
+                    aw_col_name: {"sync": sync_function},
+                    iv_col_name: {"sync": sync_function}
+                }
+            }
+        }, "num_index_replicas": 0  # This might need to change idk what vectors are
```
Comment (stanton11) [inline testsuites/CBLTester/CBL_Functional_tests/APITests/test_vectorSearch.py:66]: Does this need changing now?

File: testsuites/CBLTester/CBL_Functional_tests/APITests/test_vectorSearch.py [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+    # Check that all 4 collections on CBL exist
+    cbl_collections = db.collectionsInScope(vsTestDatabase, scope)
+    # TODO check if _default counts towards this
+    assert len(cbl_collections) == 5, "wrong number of collections returned"
+    assert dbv_col_name in cbl_collections, "no CBL collection found for doc body vectors"
+    assert st_col_name in cbl_collections, "no CBL collection found for search terms"
+    assert iv_col_name in cbl_collections, "no CBL collection found for index vectors"
+    assert aw_col_name in cbl_collections, "no CBL collection found for auxiliary words"
+
+    # replicate docs to server via sgw
```
Comment (stanton11) [inline testsuites/CBLTester/CBL_Functional_tests/APITests/test_vectorSearch.py:158]: Commented out code to be kept?

File: testsuites/CBLTester/CBL_Functional_tests/APITests/test_vectorSearch.py [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+
+    sg_docs = sg_client.get_all_docs(url=sg_admin_url, db=sg_db, auth=session, scope=scope, collection=collection)
+    sg_docs = sg_docs["rows"]
+    return len(sg_docs)
+
+
+# TODO might be worth checking if a. this test case is small enough for vector search and
+# b. whether this is an appropriate fixture for a sanity test
+# TODO make this test only pull documents from server that have embeddings then query them
+@pytest.mark.skip(reason="Waiting for all the test apps chanegs to be merged")
```
Comment (stanton11) [inline testsuites/CBLTester/CBL_Functional_tests/APITests/test_vectorSearch.py:372]: Should be "changes"

File: testsuites/CBLTester/CBL_Functional_tests/APITests/test_vectorSearch.py (hunk 2 above)
Comment (stanton11) [reply testsuites/CBLTester/CBL_Functional_tests/APITests/test_vectorSearch.py:158]: Sounds good to me!

File: testsuites/CBLTester/CBL_Functional_tests/APITests/test_vectorSearch.py (hunk 1 above)
Comment (gilkal) [reply testsuites/CBLTester/CBL_Functional_tests/APITests/test_vectorSearch.py:66]: Let's keep it as is for this phase. We might. need to change it later.

File: testsuites/CBLTester/CBL_Functional_tests/APITests/test_vectorSearch.py (hunk 3 above)
Comment (gilkal) [reply testsuites/CBLTester/CBL_Functional_tests/APITests/test_vectorSearch.py]: Thanks, fixed

**Review by stanton11 — CHANGES_REQUESTED** [review]
lgtm!

**Review by stanton11 — APPROVED** [review]
lgtm

**Pull request merged**